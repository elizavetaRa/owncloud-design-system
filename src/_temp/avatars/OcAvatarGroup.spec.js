import { shallowMount } from "@vue/test-utils"

import Group from "./OcAvatarGroup.vue"

const users = [
  {
    id: "bob",
    username: "bob",
    displayName: "Bob",
    avatar:
      "https://images.unsplash.com/photo-1610216705422-caa3fcb6d158?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTB8fGZhY2V8ZW58MHwyfDB8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=60",
  },
  {
    id: "marie",
    username: "marie",
    displayName: "Marie",
    avatar:
      "https://images.unsplash.com/photo-1584308972272-9e4e7685e80f?ixid=MXwxMjA3fDB8MHxzZWFyY2h8Mzh8fGZhY2V8ZW58MHwyfDB8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=60",
  },
  {
    id: "john",
    username: "john",
    displayName: "John Richards Emperor of long names",
  },
]

describe("OcAvatarGroup", () => {
  it("displays tooltip", async () => {
    const wrapper = shallowMount(Group, {
      propsData: {
        users,
        maxDisplayed: 2,
        isTooltipDisplayed: true,
      },
    })

    expect(wrapper.attributes()["uk-tooltip"]).toMatch("Bob, Marie +1")
    expect(wrapper).toMatchSnapshot()
  })

  it("hides tooltip if not enabled", () => {
    const wrapper = shallowMount(Group, {
      propsData: {
        users,
      },
    })

    expect(wrapper.attributes()["uk-tooltip"]).toBeFalsy()
  })
})
