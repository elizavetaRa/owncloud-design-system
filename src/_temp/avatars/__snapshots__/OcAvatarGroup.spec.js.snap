// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcAvatarGroup displays tooltip 1`] = `
<div uk-tooltip="Bob, Marie +1" class="oc-avatar-group">
  <oc-avatar-stub src="https://images.unsplash.com/photo-1610216705422-caa3fcb6d158?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTB8fGZhY2V8ZW58MHwyfDB8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=800&amp;q=60" username="bob" accessiblelabel="Bob" width="30"></oc-avatar-stub>
  <oc-avatar-stub src="https://images.unsplash.com/photo-1584308972272-9e4e7685e80f?ixid=MXwxMjA3fDB8MHxzZWFyY2h8Mzh8fGZhY2V8ZW58MHwyfDB8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=800&amp;q=60" username="marie" accessiblelabel="Marie" width="30"></oc-avatar-stub>
  <!---->
  <oc-avatar-count-stub count="1" size="30"></oc-avatar-count-stub>
</div>
`;
